File: LeetcodeProblems/blind-75/interval/meeting-rooms-ii/solution.py
# Meeting Rooms II - LeetCode #253

from typing import List
import heapq

class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        if not intervals:
            return 0
        
        intervals.sort(key=lambda x: x[0])
        
        heap = [intervals[0][1]]
        
        for interval in intervals[1:]:
            if interval[0] >= heap[0]:
                heapq.heappop(heap)
            heapq.heappush(heap, interval[1])
        
        return len(heap)

File: LeetcodeProblems/blind-75/interval/meeting-rooms-ii/solution.js
// Meeting Rooms II - LeetCode #253

/**
 * @param {number[][]} intervals
 * @return {number}
 */
var minMeetingRooms = function(intervals) {
    if (intervals.length === 0) return 0;
    
    intervals.sort((a, b) => a[0] - b[0]);
    
    const heap = [intervals[0][1]];
    
    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i][0] >= heap[0]) {
            heap[0] = intervals[i][1];
        } else {
            heap.push(intervals[i][1]);
        }
        heap.sort((a, b) => a - b); // Simulate min-heap
    }
    
    return heap.length;
};
